float32 latitude
float32 longitude
---
int8 state